subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/CompileWithAnnotationProcessingBuildOperationType.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks.compile;
+
+import org.gradle.internal.operations.BuildOperationType;
+
+import java.util.Map;
+
+public class CompileWithAnnotationProcessingBuildOperationType implements BuildOperationType<CompileWithAnnotationProcessingBuildOperationType.Details, CompileWithAnnotationProcessingBuildOperationType.Result> {
+
+    public interface Details {
+        String getTaskPath();
+    }
+
+    public interface Result {
```

Comment:
Some interesting follow-ups for this result object:

- How many files are there in total?
- How many were recompiled?
- How many were reprocessed?

Reply:
Sounds good! Shall we create a follow-up issue or wait for someone to request these?